src/Array/Extra.elm

```diff
@@ -139,7 +139,7 @@ sliceUntil lengthNew =
 -}
 pop : Array element -> Array element
 pop =
-    slice 0 -1
+    \array -> array |> slice 0 -1
```

Comment:
I really do wonder how this trick could produce better performance. Can you point me to some explanation?

Reply:
No clue. Of course a change like this only matters when that function is called repeatedly like while recursing.

Reply:
are you sure that simply avoiding the point free style like:

```elm
pop : Array element -> Array element
pop array=
    slice 0 -1 array
```

would not be same as this or faster?

Reply:
Not including the subject of the operation, the `array` argument, in the declaration is a purely stylistic choice for me
to make the preferred application style obvious: _What are the arguments_ when `|>` pipeline-ing or currying them?
I'm willing to discuss the utility of this!

Running the benchmarks on [chromium](https://github.com/elm-community/array-extra/blob/master/benchmarks/chromium.pdf) and [firefox](https://github.com/elm-community/array-extra/blob/master/benchmarks/firefox.pdf) suggests the performance is more or less identical.

Reply:
As you can see, `Array.Extra` does the same with function types:
https://github.com/elm-community/array-extra/blob/master/src/Array/Extra.elm#L97-L98
with the same reasoning

Reply:
To elaborate on the reasons, I generally assume that at every point in the code, you know so much about the to-be-transformed-subject that adding a name doesn't provide much utility that isn't already apparent from its type (and most of the time even the module prefix `Array.Extra`).

Reply:
>  is a purely stylistic choice

fine by me. I don't question stylistic choices since the only interesting thing I find about them is that they are the same thing. That's why I wondered if this is same or not in terms of performance.

I know `|>` gets optimized out (`foo |> f ` becomes `f foo`) so the only question really is if doing `f = \a -> ...` compiles to same thing as `f a = ...` or similar thing or significantly different thing.

Thanks for clarifications.

Reply:
OT: perhaps the the most elegant language design would not even give the two ways of doing the same thing. the `\arg ->` is clearly required if you want to have lambdas so `f arg =` way is the redundant one. I know about only one language which doesn't give you this option (while having currying) which is nix. There you always do `let f = x: y: z:` which is equivalent to `f = \x -> y-> z->` in elm and there is no `f x y z = ` equivalent.

Reply:
roc does the same